src/executables/util_cs.c

```diff
@@ -3260,53 +3271,64 @@ applyinfo (UTIL_FUNCTION_ARG * arg)
 	  strcat (local_database_name, "@localhost");
 
 	  db_clear_host_connected ();
-
-	  if (check_database_name (local_database_name))
+	  if ((check_applied_info != false) && check_database_name (local_database_name))
 	    {
-	      goto check_applied_info_end;
+	      printf ("\n *** Applied Info. *** \n");
+	      check_applied_info = false;
+	      PRINT_AND_LOG_ERR_MSG ("\nERROR : %s\n", db_error_string (3));
 	    }
-	  if (db_login ("DBA", NULL) != NO_ERROR)
+	  if ((check_applied_info != false) && (db_login ("DBA", NULL) != NO_ERROR))
 	    {
-	      goto check_applied_info_end;
+	      printf ("\n *** Applied Info. *** \n");
+	      check_applied_info = false;
+	      PRINT_AND_LOG_ERR_MSG ("\nERROR : %s\n", db_error_string (3));
 	    }
-	  error = db_restart (arg->command_name, TRUE, local_database_name);
-	  if (error != NO_ERROR)
+	  if ((check_applied_info != false) && (db_restart (arg->command_name, TRUE, local_database_name) != NO_ERROR))
 	    {
-	      goto check_applied_info_end;
+	      printf ("\n *** Applied Info. *** \n");
+	      check_applied_info = false;
+	      PRINT_AND_LOG_ERR_MSG ("\nERROR : %s\n", db_error_string (3));
 	    }
 
-	  if (HA_DISABLED ())
+	  if ((check_applied_info != false) && HA_DISABLED ())
```

Comment:
Similar actions and similar conditions are repeated over and over again. 
how about handling it with something like a label so that we can process them all at once?

Reply:
Good idea. But in that case, the refactoring seems to be preceded.

Reply:
The best way that we make code readable is change the order of print.

> AS-IS 
-a option + -L option 
> To-BE
-L option  + -a option

But this change is must be discussed and I think it is out of range from this issue.

Reply:
So, I can guess that you may suggest below code. Can you identify below?


```
if (check_database_name (local_database_name) && (db_login ("DBA", NULL) && (db_restart (arg->command_name, TRUE, local_database_name) )
	    {
	      printf ("\n *** Applied Info. *** \n");
	      check_applied_info = false;
	      PRINT_AND_LOG_ERR_MSG ("\nERROR : %s\n", db_error_string (3));
	    }
if ((check_applied_info != false) && HA_DISABLED ())
{
	      printf ("\n *** Applied Info. *** \n");
	      check_applied_info = false;
	      printf ("\nERROR : ");
	      PRINT_AND_LOG_ERR_MSG (msgcat_message (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_APPLYINFO,
						     APPLYINFO_MSG_NOT_HA_MODE));
	    }
if (check_applied_info == false)
	    {
	      (void) db_shutdown ();
	    }
```

Reply:
I think you can use another boolean variable that indicates whether an error occurred or not.
or
It seems to be able to define the actions to be taken when an error occurs using error label. 

whenever error occurred 1. you change the `check_applied_info`,2. print header, 3 print error message.

above 3 steps are repeated even in the code modification you suggested.

Reply:
Did you indicate two `if clauses` are repeated?

Reply:
No, sequence of actions is repeated.